src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  useEffect,
+  useState,
+  memo,
+  useMemo,
+} from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+
+import { PeopleFilters } from '../PeopleFilters';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { getFilteredPeople } from '../../utils/getFilteredPeople';
+
+export const PeoplePage: React.FC = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex') || '';
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const { slug = '' } = useParams();
+
+  useEffect(() => {
+    const fetchPeople = async () => {
+      try {
+        setIsLoading(true);
+        const peopleFromServer = await getPeople();
+
+        setPeople(peopleFromServer);
+      } catch {
+        setHasError(true);
+      } finally {
+        setIsLoaded(true);
+        setIsLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  const visiblePeople = useMemo(() => (
+    getFilteredPeople(
+      sex, centuries, query, people, sort, order,
+    )
+  ), [sex, centuries, query, people, sort, order]);
+
+  const hasPeople = isLoaded && people.length;
+  const noPeople = isLoaded && !people.length;
+  const noFilteredPeople = isLoaded && visiblePeople.length === 0;
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="columns is-desktop is-flex-direction-row-reverse">
+          {hasPeople && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters />
+            </div>
+          )}
+
+          <div className="column">
+            <div className="box table-container">
+              {isLoading && <Loader />}
+
+              {hasError && (
+                <p data-cy="peopleLoadingError">Something went wrong</p>
+              )}
+
+              {noPeople && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
+
+              {noFilteredPeople && (
+                <p>There are no people matching the current search criteria</p>
+              )}
+
+              {hasPeople && !noFilteredPeople && (
```

Comment:
```suggestion
              {hasPeople && (
```
I think that's enought